packages/component-base/test/slot-controller.test.js

```diff
@@ -120,17 +124,41 @@ describe('slot-controller', () => {
     });
 
     describe('custom element', () => {
-      beforeEach(() => {
-        element = fixtureSync(`
-          <slot-controller-element>
-            <div>baz</div>
-          </slot-controller-element>
-        `);
-        // Get element reference before adding the controller
+      let wrapper;
+
+      const initializeSpy = sinon.spy();
+
+      customElements.define(
+        'default-slot-element',
+        class extends SlotControllerHost {
+          ready() {
+            super.ready();
+
+            controller = new SlotController(this, '', 'div', { initializer: initializeSpy });
+            this.addController(controller);
+          }
+        },
+      );
+
+      customElements.define(
+        'custom-default-wrapper',
+        class extends LitElement {
+          render() {
+            return html`
+              <default-slot-element>
+                <div>baz</div>
+              </default-slot-element>
```

Comment:
I'm guess this is the part that covers the empty text node scenario, but I'm not sure. Should we make it more obvious with a separate test case or a comment?

Reply:
I can add a comment explaining that empty text nodes are added by Lit and get picked up by the observer.

Reply:
I'm thinking that it the test setup just incidentally works because of the code formatting in the template. Adding a HTML comment could help enforcing a newline somewhere in the template.

Reply:
Good point, added the comment in the code and also HTML comments in the fixture.